core/src/ecs/systems/ui/InputSystem.ts

```diff
@@ -175,7 +176,7 @@ export const InputSystem = ClientSystemBuilder({
           // check for single key pressed
           const controllerInput = input.inputMap.press[keyPress];
           if (controllerInput != null) {
-            const invocation = controllerInput({ mouse, entity: character, world, tick: keyMouse.tick });
+            const invocation = controllerInput({ mouse: { ...mouse }, entity: character, world, tick: keyMouse.tick });
```

Comment:
have to do this spread `mouse: { ...mouse }` to copy the object state at that time